<?xml version="1.0"?>

<!--FIX LIMIT AND THE NOMENCLATURE: JOINTS AND LINK NUMBER IS OFF-->
<!--FIX THE MODEL DIMENSION: FOR LINK BASE AND LINK1-->
<!--ADD INERTIA FOR ALL LINKS-->
<!--REVISE DH PARAMETERS-->
<!-- REDO LINKS 6 and 7 -->
<!--  -->
<!--USE XARCO AND TRANSFER TO SRDF FOR GAZEBO-->
<robot name="roboto_visual" xmlns:xacro="http://ros.org/wiki/xacro">
	<xacro:property name="base_inertial">
		<inertial>
      		<origin
        		xyz="-0.11257 -4.1589E-05 0.057533"
        		rpy="0 0 0" />
      		<mass
        		value="25.417" />
      		<inertia
        		ixx="0.347632032016038"
        		ixy="-0.000497937869254212"
        		ixz="0.0371909126715644"
        		iyy="0.710520929572187"
        		iyz="0.000244473496051384"
        		izz="0.978461123632107" />
    	</inertial>
	</xacro:property>

	<xacro:property name="link1_inertial">
		<inertial>
			<origin
                xyz="0.048532 0.0047642 0.2152"
                rpy="0 0 0" />
            <mass value="32.007" />
            <inertia
                ixx="1.06695162498236"
                ixy="0.0263563429413544"
                ixz="-0.227777437748705"
                iyy="0.810219541810416"
                iyz="0.00112484567476821"
                izz="0.977594438459942" />
		</inertial>
	</xacro:property>

	<xacro:property name="link2_inertial">
		<inertial>
			<origin
        		xyz="0.26821 -0.026226 -0.00043323"
        		rpy="0 0 0" />
			<mass
        		value="7.6192" />
      		<inertia
        		ixx="0.0182518895695961"
        		ixy="0.000288617141409028"
        		ixz="-0.000801408670696314"
        		iyy="0.282929525631837"
        		iyz="0.000419364871130956"
        		izz="0.283301435599079" />
		</inertial>
	</xacro:property>

	<xacro:property name="link3_inertial">
		<inertial>
			<origin
        		xyz="0.032316 -0.10506 -0.027437"
        		rpy="0 0 0" />
      		<mass
        		value="7.4487" />
      		<inertia
        		ixx="0.0559221269064628"
        		ixy="0.00737491499245027"
        		ixz="0.00817092524689219"
        		iyy="0.0796970494385453"
        		iyz="-0.00512048798704862"
        		izz="0.0723378108896853" />
		</inertial>
	</xacro:property>

	<material name="orange">
		<color rgba="0.756 0.3608 0.0392 1"/>
	</material>

	<material name="silver">
		<color rgba="0.6157 0.8392 0.9411 1"/>
	</material>

	<link name="world"/>

	<link name="base">
		<visual>
			<geometry>
				<mesh filename="package://urdf_models/meshes/base.STL"/>
			</geometry>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<material name="orange"/>
		</visual>
	</link>

	<link name="link1">
		<visual>
			<geometry>
				<mesh filename="package://urdf_models/meshes/link1.STL"/>
			</geometry>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<material name="orange"/>
		</visual>
	</link>

	<joint name="world_to_base" type="fixed">
		<parent link="world"/>
		<child link="base"/>
		<origin xyz="0 0 0"/>
	</joint>

	<joint name="joint1" type="revolute">
		<axis xyz="0 0 1"/>
		<limit effort="1.0" lower="-1.4835" upper="1.4835" velocity="5.0"/>
		<origin rpy="0 0 0" xyz="0 0 0.118"/>
		<parent link="base"/>
		<child link="link1"/>
	</joint>
	
	<link name="link2">
		<visual>
			<geometry>
				<mesh filename="package://urdf_models/meshes/link2.STL"/>
			</geometry>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<material name="orange"/>
		</visual>
	</link>
	
	<joint name="joint2" type="revolute">
		<axis xyz="0 0 1"/>
		<limit effort="1.0" lower="-2.7925" upper="0" velocity="5.0"/>
		<!--LIMITS AND ORIGIN NEED CORRECTION -->
		<origin rpy="-1.5707 0 0" xyz="0.15 0 0.345"/>
		<parent link="link1"/>
		<child link="link2"/>
	</joint>

	<link name="link3">
		<visual>
			<geometry>
				<mesh filename="package://urdf_models/meshes/link3.STL"/>
			</geometry>
			<origin xyz="0 0 0" rpy="0 0 1.5707"/>
			<material name="orange"/>
		</visual>
	</link>

	<joint name="joint3" type="revolute">
		<axis xyz="0 0 1"/>
		<!-- LIMITS AND ORIGIN NEED CORRECTION -->
		<limit effort="1.0" lower="-1.5708" upper="1.5708" velocity="5.0"/>
		<origin rpy="0 0 0" xyz="0.6 0 0"/>
		<parent link="link2"/>
		<child link="link3"/>
	</joint>

	<link name="link4">
		<visual>
			<geometry>
				<mesh filename="package://urdf_models/meshes/link4_2nd.STL"/>
			</geometry>
			<origin xyz="0 0 0" rpy="1.5708 0 0"/>
			<material name="silver"/>
		</visual>
	</link>
	<joint name="joint4" type="revolute">
		<axis xyz="0 0 1"/>
		<limit effort="1.0" lower="0" upper="3.142" velocity="5.0"/>
		<origin rpy="-1.5707 0 0" xyz="0.12 0 0"/>
		<parent link="link3"/>
		<child link="link4"/>
	</joint>
	
	<link name="link5">
		<visual>
			<geometry>
				<mesh filename="package://urdf_models/meshes/link5.STL"/>
			</geometry>
			<origin xyz="0 0 0" rpy="0 0 -1.5708"/>
			<material name="orange"/>
		</visual>
	</link>
	<joint name="joint5" type="revolute">
		<axis xyz="0 0 1"/>
		<limit effort="1.0" lower="0" upper="3.142" velocity="5.0"/>
		<origin rpy="1.5708 0 0" xyz="0 0 0.72"/>
		<parent link="link4"/>
		<child link="link5"/>
	</joint>
	
	<link name="link6">
		<visual>
			<geometry>
				<mesh filename="package://urdf_models/meshes/link6.STL"/>
			</geometry>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<material name="silver"/>
		</visual>
	</link>
	<!--LINK6 IS A DUMMY LINK TO RECONCILE THE LAST JOINT WITH THE END EFFECTOR-->
	<joint name="joint6" type="revolute">
		<axis xyz="0 0 1"/>
		<limit effort="1.0" lower="0" upper="3.142" velocity="5.0"/>
		<origin rpy="-1.5708 0 0" xyz="0 0 0"/>
		<parent link="link5"/>
		<child link="link6"/>
	</joint>
	
	<link name="link7">
		<!--visual>
			<geometry>
				<mesh filename="package://urdf_models/meshes/link6.STL"/>
			</geometry>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<material name="silver"/>
		</visual-->
	</link>
	<joint name="joint7" type="fixed">
		<!--axis xyz="0 0 1"/>
		<limit effort="1.0" lower="0" upper="3.142" velocity="5.0"/-->
		<origin rpy="0 0 0" xyz="0 0 0.085"/>
		<parent link="link6"/>
		<child link="link7"/>
	</joint>


</robot>
